sharepoint-adfs/azuredeploy.json

```diff
@@ -179,7 +209,8 @@
       "spSvcPassword": "[parameters('serviceAccountsPassword')]",
       "spAppPoolUserName": "spapppool",
       "spAppPoolPassword": "[parameters('serviceAccountsPassword')]",
-      "spPassphrase": "[parameters('serviceAccountsPassword')]"
+      "spPassphrase": "[parameters('serviceAccountsPassword')]",
+      "baseUri": "[if(empty(parameters('_artifactsLocation')), deployment().properties.templateLink.uri, parameters('_artifactsLocation'))]"
```

Comment:
remove this and just use:

[deployment().properties.templateLink.uri]

As the defaultValue on the parameter...

Reply:
@bmoore-msft Assuming I understood your request correctly, I think this change is not compatible with Travis test because property [templateLink](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-deployment) is available only if template is deployed through the portal.
It also prevents my own deployments when I use cmdlet New-AzResourceGroupDeployment.
I'll revert the change and await for your feedback.

Reply:
Here is a related issue: https://github.com/MicrosoftDocs/azure-docs/issues/21584

Reply:
I'm just commenting on your use of if() to replace an empty parameter.  In those cases, you can just use a default parameter.  If it's not working in some scenario, it's a different problem/bug.

Reply:
I added the if to be able to use deployment().properties.templateLink.uri when possible, as it does not work in some scenarios as described in the [contribution guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts)
I just pushed the following changes:

* Completely removed use of deployment().properties.templateLink.uri (which I was hoping to keep) and my intermediate variable 'baseUri'
* Updated DSC URIs to be exactly like in the contribution guide